### PR TITLE
[core] Extend ActorPool API to support adding / removing actors 

### LINKED ISSUE #13704: Adding / removing actors from actor pool
### Describe your feature request

It would be great if we can able to add/remove actors from actor pool dynamically in middle of the program.  Can you add transferring idle actors from one pool to another pool dynamically inside the program 

For feature requests or questions, post on our Discussion page instead: https://discuss.ray.io/

### PR DESCRIPTION
**Pull request by micahtyong**
## In this PR 

Extends the `ActorPool` API by exposing three new functions from the `ActorPool` class: 
- `pop_idle()`: removes the first idle actor from a pool
- `push()`: adds an actor to a pool 
- `has_free()`: returns whether or not an idle actor exists in a pool 

To use this API, you can do the following:  
```
pool1 = ActorPool([a1, a2])
pool2 = ActorPool([b1])
# after finishing up work for pool1 ...
while pool1.has_free(): 
    pool2.push(pool1. pop_idle())
```

## Related issue number

Closes #13704. 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/util/actor_pool.py [hunk 1]
```diff
@@ -213,3 +213,64 @@ def _return_actor(self, actor):
...
+            >>> print(pool.has_free())
+            False
+            >>> print(pool.get_next())
+            2
+            >>> print(pool.has_free())
+            True
+        """
+        return len(self._idle_actors) > 0 and len(self._pending_submits) == 0
+
+    def pop(self):
```
Comment (wuisawesome) [inline python/ray/util/actor_pool.py:236]: Can we rename this `pop_idle`?

Comment (micahtyong) [reply python/ray/util/actor_pool.py]: Sure thing! I agree this makes the function more clear. Just changed!

**Review by wuisawesome — APPROVED** [review]

**Comment by rkooo567** [comment]
CAn you merge the lastest master?

**Comment by micahtyong** [comment]
Done! @rkooo567

**Comment by rkooo567** [comment]
Rerunning test_actor in Windows because it could be related (not 100% sure if test actor contains actor pool test though)

**Comment by wuisawesome** [comment]
Windows is passing now and mac build failure looks unrelated. Merging

**Pull request merged**